### PR TITLE
Add support for sending queries to remote nodes

### PR DESCRIPTION
**Pull request by philss**
It uses `:rpc` to send queries to remote nodes.
The syntax is similar to the used by `GenServer`, where it accepts a tuple with `{repo, node}`.

This PR also includes some instructions for development and a fix to the CI configuration.
I can split the PR in two if you like :)

PS: this is related to https://github.com/phoenixframework/phoenix_live_dashboard/pull/327 and aims to support distribution in Phoenix Live Dashboard Ecto's page.

### CODE REVIEW

**Comment by philss** [comment]
ping @josevalim :)

**Comment by philss** [comment]
Just realize that `GenServer` uses `{atom, node}` instead of `{node, atom}` adopted here. Do you think we should change here?

**Comment by josevalim** [comment]
@philss yes, we should go with the GenServer style (I always get confused about it).

**Comment by josevalim** [comment]
@philss yes, we should go with the GenServer style (I always get confused about it).

File: test/support/distribution_support.ex [hunk 1]
```diff
@@ -0,0 +1,57 @@
+defmodule EctoPSQLExtras.DistributionSupport do
+  # This module helps to spawn a new node to test
+  # distribution. The remote node does not requires
+  # EctoPSQLExtras, but requires that the repository
+  # exists.
+  #
+  # Check the "test/support/dummy_app.exs".
+  # It requires Elixir 1.12
+  def setup_support_project!(filename) do
```
Comment (josevalim) [inline test/support/distribution_support.ex]: Erlang actually has a functionality to build this. It is in the very poorly named module called `slave`. Here is how Phoenix PubSub uses it to spawn new nodes: https://github.com/phoenixframework/phoenix_pubsub/blob/master/test/support/cluster.ex#L1

**Comment by pawurb** [comment]
Hi. Thanks for the PR! I'm currently not actively coding in Elixir, so I'll leave the review up to @josevalim and merge it as soon as there's a 👍

File: test/support/distribution_support.ex (hunk 1 above)
Comment (philss) [reply test/support/distribution_support.ex]: @josevalim Cool! And yeah, poorly named indeed :sweat: 

But I think is not quite the same I want here. My idea is to start a node without the Ecto PSQL Extra code, but with Ecto and a simple repository.
Do you think this is too much? I mean: should I only focus on having a remote node with the same code instead?

Comment (josevalim) [reply test/support/distribution_support.ex]: You can control exactly what is loaded on the other node. Just make sure you remove ecto_psql stuff from the code path here: https://github.com/phoenixframework/phoenix_pubsub/blob/master/test/support/cluster.ex#L18

Comment (philss) [reply test/support/distribution_support.ex]: Oh, I see! I will try that later today. Thanks! :D

Comment (philss) [reply test/support/distribution_support.ex]: @josevalim Done! Sorry for the delay

**Review by josevalim — APPROVED** [review]
LGTM! @pawurb, can you please merge and do a new release when all tests pass and you agree? Thank you <3

**Comment by philss** [comment]
Sorry about the failures. I'm not sure why it doesn't work on CI.
I changed to use `Node.self()` and it should work now.

**Comment by pawurb** [comment]
Thanks, I'll be releasing as 0.7.0.

**Comment by philss** [comment]
@pawurb thank you! and thanks @josevalim for the review :purple_heart:

**Pull request merged**